src/main.rs

```diff
@@ -68,11 +103,57 @@ async fn main() -> Result<()> {
         warn!("INSECURE: The security of Keylime is NOT linked to a hardware root of trust.");
         warn!("INSECURE: Only use Keylime in this mode for testing or debugging purposes.");
     }
+
+    // Gather EK and AK key values and certs
+    let (ek_handle, ek_cert, ek_tpm2b_pub) =
+        tpm::create_ek(&mut ctx, Some(AsymmetricAlgorithm::Rsa))?;
+
+    let (ak_handle, ak_name, ak_tpm2b_pub) =
+        tpm::create_ak(&mut ctx, ek_handle)?;
+
+    // Gather configs
     let cloudagent_ip =
         config_get("/etc/keylime.conf", "cloud_agent", "cloudagent_ip")?;
     let cloudagent_port =
         config_get("/etc/keylime.conf", "cloud_agent", "cloudagent_port")?;
-    info!("Starting server...");
+    let registrar_ip =
+        config_get("/etc/keylime.conf", "registrar", "registrar_ip")?;
+    let registrar_port =
+        config_get("/etc/keylime.conf", "registrar", "registrar_port")?;
+    let agent_uuid_config =
+        config_get("/etc/keylime.conf", "cloud_agent", "agent_uuid")?;
+    let agent_uuid = get_uuid(&agent_uuid_config);
+
+    {
```

Comment:
@puiterwijk What is the purpose of this bracket? I believe this is from your changes.

Reply:
that does not look like a strange scope, is the compiler not complaining about this?

Reply:
This was basically because this activation step is in the `main()` function, which keeps running until the entire agent terminates.
And I didn't want to have secrets lingering in memory during the full runtime.